File: 1381DesignAStackWithIncrementOperation.py
class CustomStack:

    def __init__(self, maxSize: int):
        self.arr = []
        self.maxSize = maxSize

    def push(self, x: int) -> None:
        if len(self.arr) < self.maxSize:
            self.arr.append(x)

    def pop(self) -> int:
        if len(self.arr) == 0:
            return -1
        else:
            pop_num = self.arr[len(self.arr) - 1]
            self.arr.pop(len(self.arr) - 1)
            return pop_num


    def increment(self, k: int, val: int) -> None:
        if k > len(self.arr):
            for i in range(0, len(self.arr)):
                self.arr[i] += val
        else:
            for i in range(0, k):
                self.arr[i] += val
    


# Your CustomStack object will be instantiated and called as such:
# obj = CustomStack(maxSize)
# obj.push(x)
# param_2 = obj.pop()
# obj.increment(k,val)

File: 1381DesignAStackWithIncrementOperation.cpp
class CustomStack {
public:
    vector<int> arr;
    int capacity;

    CustomStack(int maxSize) {
        capacity = maxSize;
    }
    
    void push(int x) {
        if (int(arr.size()) < capacity)
            arr.push_back(x);
    }
    
    int pop() {
        if (arr.size() == 0)
            return -1;
        else{
            int pop_num = arr[arr.size() -1];
            arr.pop_back();
            return pop_num;
        }
    }
    
    void increment(int k, int val) {
        if (k > arr.size()){
            for (int i = 0 ; i < arr.size() ; i++){
                arr[i] += val;
            }
        }
        else{
            for (int i = 0 ; i < k ; i++){
                arr[i] += val;
            }
        }
    }
};

/**
 * Your CustomStack object will be instantiated and called as such:
 * CustomStack* obj = new CustomStack(maxSize);
 * obj->push(x);
 * int param_2 = obj->pop();
 * obj->increment(k,val);
 */
